hist/hist/inc/TH1.h

```diff
@@ -673,7 +677,11 @@ class TH1D : public TH1, public TArrayD {
    TH1D& operator=(const TH1D &h1);
    ~TH1D() override;
 
-   void     AddBinContent(Int_t bin) override {++fArray[bin];}
+   /// Increment bin content by 1.
+   /// Passing an out-of-range bin leads to undefined behavior
+   void     AddBinContent(Int_t bin) override {if (fArray[bin] < 9007199254740992.) ++fArray[bin];}
```

Comment:
I would not add any check for double histograms.
This is the default histogram in ROOT and we want to keep max efficiency

Reply:
Makes sense. Although I'd rather say that the default histogram in ROOT is TH1F. See TTree::Draw.